### PR TITLE
Simplify account view by using helpers

### LINKED ISSUE #929: Simplify entity views - number_to_currency rendering
In a number of places, our templates have a fair amount of display logic:
https://github.com/fatfreecrm/fat_free_crm/blob/master/app/views/accounts/_sidebar_show.html.haml#L29

```
pipeline != 0.0 ? number_to_currency(pipeline, precision: 0) : t(:n_a)
won != 0.0 ? number_to_currency(won, precision: 0) : t(:n_a)
lost != 0.0 ? number_to_currency(lost, precision: 0) : t(:n_a)
```

We should look at either a very generic helper; or more specifically pulling these out to an accounts_helper and using rubyisms like .zero? so that our templates are basically html element, plus variable.

### PR DESCRIPTION
**Pull request by johnbumgardner**
Closes #929 
I moved some of the logic on how to display pipeline, won, and lost into accounts_helper.rb and wrote RSpec tests for these new functions.

### CODE REVIEW

**Comment by johnbumgardner** [comment]
@CloCkWeRX are there any other views that could use refactoring?

File: app/helpers/accounts_helper.rb [hunk 1]
```diff
@@ -104,4 +104,26 @@ def brief_account_info(contact)
         end
     text.html_safe
   end
+
+  # We have too much login in the account views
```
Comment (CloCkWeRX) [inline app/helpers/accounts_helper.rb:108]: logic?

File: app/helpers/accounts_helper.rb [hunk 2]
```diff
@@ -104,4 +104,26 @@ def brief_account_info(contact)
...
+  # - a helper that abstracts the logic to the backend
+  def display_pipeline(pipeline)
```
Comment (CloCkWeRX) [inline app/helpers/accounts_helper.rb:110]: These might be better merged into a single function, as they repeat the same logic a few times.x

**Review by CloCkWeRX — APPROVED** [review]
Its certainly a lot easier to understand. Will leave it up to you if you want to merge the repeated code.

File: app/helpers/accounts_helper.rb (hunk 2 above)
Comment (johnbumgardner) [reply app/helpers/accounts_helper.rb]: pipeline, won, and lost are all one function now

**Comment by akeemcloud1** [comment]
Hi guys.

kindly help on publish, the step by step guide on how to set up FatfreeCRM project on Ubuntu Linux.

I just got stuck at rake:db:create since weeks now.

awaiting response pleaseeeeeeeeeeeeeeeeeeeeeeeeee!

**Comment by johnbumgardner** [comment]
@akeemcloud1 
https://github.com/fatfreecrm/fat_free_crm/wiki/Setup-Linux-or-Mac-OS
Are these steps insufficient?

**Comment by akeemcloud1** [comment]
john.bumgardner@icloud.com its not sufficient enough

**Comment by akeemcloud1** [comment]
Invoke db:create (first_time)
** Invoke db:load_config (first_time)
** Invoke environment (first_time)
** Execute environment
rake aborted!
NoMethodError: undefined method `arity' for {:as=>:commentable,
:dependent=>:destroy}:Hash
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activerecord-6.0.3.7/lib/active_record/associations/builder/association.rb:51:in
`build_scope'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activerecord-6.0.3.7/lib/active_record/associations/builder/association.rb:45:in
`create_reflection'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activerecord-6.0.3.7/lib/active_record/associations/builder/association.rb:30:in
`build'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activerecord-6.0.3.7/lib/active_record/associations.rb:1371:in
`has_many'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/acts_as_commentable-4.0.2/lib/commentable_methods.rb:58:in
`acts_as_commentable'
/home/appuser/prod/fat_free_crm/app/models/entities/lead.rb:66:in
`<class:Lead>'
/home/appuser/prod/fat_free_crm/app/models/entities/lead.rb:41:in `<main>'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in
`block in require_with_bootsnap_lfi'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in
`register'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in
`require_with_bootsnap_lfi'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:282:in
`const_get'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:282:in
`block in constantize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:280:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:280:in
`inject'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:280:in
`constantize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/core_ext/string/inflections.rb:68:in
`constantize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/active_model/observing.rb:309:in
`block in observe'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/active_model/observing.rb:309:in
`collect!'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/active_model/observing.rb:309:in
`observe'
/home/appuser/prod/fat_free_crm/app/models/observers/lead_observer.rb:9:in
`<class:LeadObserver>'
/home/appuser/prod/fat_free_crm/app/models/observers/lead_observer.rb:8:in
`<main>'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in
`block in require_with_bootsnap_lfi'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in
`register'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in
`require_with_bootsnap_lfi'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:282:in
`const_get'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:282:in
`block in constantize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:280:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:280:in
`inject'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/inflector/methods.rb:280:in
`constantize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/core_ext/string/inflections.rb:68:in
`constantize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/active_model/observing.rb:183:in
`instantiate_observer'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/active_model/observing.rb:108:in
`block in instantiate_observers'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/active_model/observing.rb:108:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/active_model/observing.rb:108:in
`instantiate_observers'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/railtie.rb:42:in
`block (2 levels) in <class:Railtie>'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:71:in
`class_eval'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:71:in
`block in execute_hook'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:61:in
`with_execution_control'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:66:in
`execute_hook'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:43:in
`block in on_load'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:42:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:42:in
`on_load'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rails-observers-0.1.5/lib/rails/observers/railtie.rb:41:in
`block in <class:Railtie>'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:68:in
`block in execute_hook'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:61:in
`with_execution_control'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:66:in
`execute_hook'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:52:in
`block in run_load_hooks'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:51:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/lazy_load_hooks.rb:51:in
`run_load_hooks'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/application/finisher.rb:129:in
`block in <module:Finisher>'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/initializable.rb:32:in
`instance_exec'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/initializable.rb:32:in
`run'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/initializable.rb:61:in
`block in run_initializers'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:228:in `block in
tsort_each'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:350:in `block (2
levels) in each_strongly_connected_component'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:431:in
`each_strongly_connected_component_from'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:349:in `block in
each_strongly_connected_component'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:347:in `each'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:347:in `call'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:347:in
`each_strongly_connected_component'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
/usr/share/rvm/rubies/ruby-3.0.0/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/initializable.rb:60:in
`run_initializers'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/application.rb:363:in
`initialize!'
/home/appuser/prod/fat_free_crm/config/environment.rb:12:in `<main>'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in
`block in require_with_bootsnap_lfi'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in
`register'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in
`require_with_bootsnap_lfi'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in
`block in require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:291:in
`load_dependency'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/activesupport-6.0.3.7/lib/active_support/dependencies.rb:324:in
`require'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/application.rb:339:in
`require_environment!'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/railties-6.0.3.7/lib/rails/application.rb:523:in
`block in run_tasks_blocks'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in
`block in execute'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:281:in
`execute'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:219:in
`block in invoke_with_call_chain'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in
`synchronize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in
`invoke_with_call_chain'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:243:in
`block in invoke_prerequisites'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:241:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:241:in
`invoke_prerequisites'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:218:in
`block in invoke_with_call_chain'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in
`synchronize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in
`invoke_with_call_chain'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:243:in
`block in invoke_prerequisites'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:241:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:241:in
`invoke_prerequisites'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:218:in
`block in invoke_with_call_chain'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in
`synchronize'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:199:in
`invoke_with_call_chain'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/task.rb:188:in
`invoke'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:160:in
`invoke_task'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in
`block (2 levels) in top_level'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in
`each'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:116:in
`block in top_level'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:125:in
`run_with_threads'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:110:in
`top_level'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:83:in
`block in run'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:186:in
`standard_exception_handling'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/lib/rake/application.rb:80:in
`run'
/home/appuser/.rvm/gems/ruby-3.0.0/gems/rake-13.0.3/exe/rake:27:in `<top
(required)>'
/home/appuser/.rvm/gems/ruby-3.0.0/bin/rake:23:in `load'
/home/appuser/.rvm/gems/ruby-3.0.0/bin/rake:23:in `<main>'
Tasks: TOP => db:create => db:load_config => environment


i got this, and i wish to move on...

**Comment by steveyken** [comment]
Looks like you are using ruby 3 there. I suspect (though I haven't tried it
myself) that Fat Free CRM isn't quite ready for that at the moment.

Would suggest using ruby 2.7 to get going.

**Comment by akeemcloud1** [comment]
what are the things i need to check out, if i run  rake db:create, rake db:migrate

cos running into issues, you guys should help out please.

Am new to this apps!

**Comment by steveyken** [comment]
Have you had a look at the installation documents? They should have the
details you need.

https://github.com/fatfreecrm/fat_free_crm

This particular thread is related more to account view helpers code. Please
create another ticket if you encounter further errors. Thanks!

**Pull request merged**